test/UnitLoadTorture.cpp

```diff
@@ -76,41 +81,35 @@ int UnitLoadTorture::loadTorture(const std::string& name, const std::string& doc
 
                 int viewid = -1;
                 LOK_ASSERT(COOLProtocol::getTokenIntegerFromMessage(status, "viewid", viewid));
-                sum_view_ids += viewid;
-                ++num_of_views;
-                --num_to_load;
 
-                TST_LOG(": #" << id << ", loaded views: " << num_of_views
-                              << ", to load: " << num_to_load);
+                LOK_ASSERT("Failed to create view in time " && viewid >= 0);
 
-                while (true)
                 {
-                    if (num_to_load == 0)
-                    {
-                        // Unload at once, nothing more left to do.
-                        TST_LOG(": #" << id << ", no more to load, unloading.");
-                        break;
-                    }
-
-                    const auto ms
-                        = (max_jitter_ms > 0
-                               ? std::chrono::milliseconds(Util::rng::getNext() % max_jitter_ms)
-                               : std::chrono::milliseconds(0));
-                    std::this_thread::sleep_for(ms);
-
-                    // Unload only when we aren't the last/only.
-                    if (--num_of_views > 0)
-                    {
-                        TST_LOG(": #" << id << ", views: " << num_of_views
-                                      << " not the last/only, unloading.");
-                        break;
-                    }
-                    else
-                    {
-                        // Correct back, since we aren't unloading just yet.
-                        ++num_of_views;
-                    }
+                    std::lock_guard<std::mutex> guard(stateLock);
+                    LOK_ASSERT("Duplicate view-id generated " && std::find(view_ids.begin(), view_ids.end(), viewid) == view_ids.end());
+                    view_ids.push_back(viewid);
                 }
+
+                ++num_of_views;
+                --num_to_load;
+
+                TST_LOG(": #" << id << ", new viewId: " << viewid <<
+                        ", loaded views: " << num_of_views <<
+                        ", to load: " << num_to_load);
+
+                // Get all the views loaded - lean on test timeout to interrupt.
+                while (num_to_load > 0)
+                    std::this_thread::sleep_for(std::chrono::milliseconds(20));
+
+                // delay randomly to close in random order:
+                const auto ms
+                    = (max_jitter_ms > 0
+                       ? std::chrono::milliseconds(Util::rng::getNext() % max_jitter_ms)
+                       : std::chrono::milliseconds(0));
```

Comment:
Randomness is generally a bad idea in tests because it affects reproducibility. Obviously there is some inherent randomness because it's multithreaded, but if we can avoid adding more that would be nice.

Edit: Actually I don't see where this value is used at all

Reply:
Good point we're missing a sleep =) Of course cleaning this up to have a nice fixed order that is pathological would be great but ...